src/System Application/App/AppSource Gallery/src/AppSourceProductDetails.Page.al

```diff
@@ -0,0 +1,395 @@
+        PlanLinePrUserPrMonthLbl: Label '%1 %2 user/month', Comment = 'Price added a plan line, %1 is the currency, %2 is the price';
+        PlanLinePrUserPrYearLbl: Label '%1 %2 user/year', Comment = 'Price added a plan line, %1 is the currency, %2 is the price';
+        PlanLineFirstMonthIsFreeLbl: Label 'First month free', Comment = 'Added to the plan line when the first month is free.';
+        PlanLinePostFillerIfFreeLbl: Label ', then ', Comment = 'Added to the plan line when the first month is free.';
+        PlanLinePriceVariesLbl: Label 'Varies', Comment = 'Added to the plan line when the price varies.';
+        PlanLinesTemplateLbl: Label '<table width="100%" padding="2" style="border-collapse:collapse;text-align:left;vertical-align:top;"><tr style="border-bottom: 1pt solid black;"><td>%1</td><td>%2</td><td>%3</td><td>%4</td></tr>%5</table>', Comment = 'Template for the plans section, %1 is the plans column header, %2 is the description column header, %3 is the monthly price column header, %4 is the yearly column header, %5 is the plan rows', Locked = true;
+        PlanLineItemTemplateLbl: Label '<tr style="text-align:left;vertical-align:top;"><td>%1</td><td>%2</td><td>%3</td><td>%4</td></tr>', Comment = 'Template for a plan line item, %1 is the plan name, %2 is the plan description, %3 is the monthly price, %4 is the annual price', Locked = true;
+        PlanLinesColumnPlansLbl: Label 'Plans', Comment = 'Column header for the plans section';
+        PlanLinesColumnDescriptionLbl: Label 'Description', Comment = 'Column header for the plans section';
+        PlanLinesColumnMonthlyPriceLbl: Label 'Monthly Price', Comment = 'Column header for the plans section';
+        PlanLinesColumnAnnualPriceLbl: Label 'Annual Price', Comment = 'Column header for the plans section';
+
+    procedure SetProduct(var ToProductObject: JsonObject)
+    var
+        ProductPlansToken: JsonToken;
+    begin
+        ProductObject := ToProductObject;
+        UniqueProductID := AppSourceProductManager.GetStringValue(ProductObject, 'uniqueProductId');
+        AppId := AppSourceProductManager.ExtractAppIDFromUniqueProductID(UniqueProductID);
+        CurrentRecordCanBeUninstalled := false;
+        CurrentRecordCanBeInstalled := false;
+        if (AppID <> '') then
+            CurrentRecordCanBeUninstalled := ExtensionManagement.IsInstalledByAppID(AppID);
+
+        if ProductObject.Get('plans', ProductPlansToken) then
+            RenderPlans(ProductPlansToken);
+    end;
+
+    procedure RenderPlans(PlansObject: JsonToken)
+    var
+        AllPlans: JsonArray;
+        PlanLinesBuilder: TextBuilder;
+        PlanItem: JsonToken;
+        PlanItemObject: JsonObject;
+        PlanItemArray: JsonArray;
+        MonthlyPriceText, YearlyPriceText : Text;
+        i, availabilitiesAdded : Integer;
+    begin
+        availabilitiesAdded := 0;
```

Comment:
no need to initialize/reset local variables

Reply:
Codecop AA0205 would disagree with you :-)

https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/analyzers/codecop-aa0205

In this case AA0205 would not trigger because the first use of the variable is += 1 which i think counts as an initialization.

The rule isn't as much about initializing, which in .al is not a thing, but more about not forgetting to set a value before using the variable... yes, it happens...